publish-pypi.yml

```diff
@@ -0,0 +1,123 @@
+        pip install cibuildwheel==0.11.1
+        cibuildwheel --output-dir wheelhouse .
+    - task: PublishPipelineArtifact@0
+      inputs: {targetPath: 'wheelhouse', artifactName: 'macos_wheels' }
+
+- job: windows_wheels
+  dependsOn: ${{ parameters.dependsOn }}
+  condition: succeeded()
+  pool: {vmImage: 'vs2017-win2016'}
+  steps:
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '3.6', architecture: x64}}
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '3.7', architecture: x64}}
+    - bash: |
+        python -m pip install --upgrade pip
+        pip install cibuildwheel==0.11.1
+        cibuildwheel --output-dir wheelhouse .
+    - task: PublishPipelineArtifact@0
+      inputs: {targetPath: 'wheelhouse', artifactName: 'windows_wheels' }
+
+- job: sdist
+  dependsOn: ${{ parameters.dependsOn }}
+  condition: succeeded()
+  pool: {vmImage: 'Ubuntu-16.04'}
+  steps:
+  - script: |
+      # When you paste this, please make sure the indentation is preserved
+      # Fail out if any setups fail
+      set -e
+
+      # Delete old Pythons
+      rm -rf $AGENT_TOOLSDIRECTORY/Python/2.7.16
+      rm -rf $AGENT_TOOLSDIRECTORY/Python/3.5.7
+      rm -rf $AGENT_TOOLSDIRECTORY/Python/3.7.3
+
+      # Download new Pythons
+      azcopy --recursive \
+        --source https://vstsagenttools.blob.core.windows.net/tools/hostedtoolcache/linux/Python/2.7.15 \
+        --destination $AGENT_TOOLSDIRECTORY/Python/2.7.15
+
+      azcopy --recursive \
+        --source https://vstsagenttools.blob.core.windows.net/tools/hostedtoolcache/linux/Python/3.5.5 \
+        --destination $AGENT_TOOLSDIRECTORY/Python/3.5.5
+
+      azcopy --recursive \
+        --source https://vstsagenttools.blob.core.windows.net/tools/hostedtoolcache/linux/Python/3.7.2 \
+        --destination $AGENT_TOOLSDIRECTORY/Python/3.7.2
+
+      # Install new Pythons
+      original_directory=$PWD
+      setups=$(find $AGENT_TOOLSDIRECTORY/Python -name setup.sh)
+      for setup in $setups; do
+          chmod +x $setup;
+          cd $(dirname $setup);
+          ./$(basename $setup);
+          cd $original_directory;
+      done;
+    displayName: 'Workaround: roll back Python versions'
+  - task: UsePythonVersion@0
+    displayName: setup python3.7
+    inputs: {versionSpec: '3.7'}
+  # Remove the ubuntu-toolchain-r/test PPA, which is added by default. Some
+  # packages were removed, and this is causing the g++multilib install to fail.
+  # Similar issue: https://github.com/scikit-learn/scikit-learn/issues/13928
+  - bash: 'sudo add-apt-repository --remove ppa:ubuntu-toolchain-r/test'
+    displayName: "Remove ppa:ubuntu-toolchain-r/test"
+  - script: 'sudo apt-get install -y --fix-missing libblas-dev liblapack-dev'
+    displayName: "Install Scipy deps"
+  - script: 'python -m pip install -U --user --force-reinstall pep517 setuptools_scm'
+    displayName: "acquire build tools"
+  - script: 'python -m pep517.build --source --out-dir wheelhouse .'
```

Comment:
Is it worth explicitly updating the submodule version before here? Or are we always sure we are getting a clean checkout?

Reply:
It should be a clean machine that clones and checkouts the submodule.

Reply:
Would you prefer we delete the submodule or manually init it?

Reply:
it should be fine in this context.